src/coreclr/System.Private.CoreLib/src/System/Reflection/TypeNameParser.CoreCLR.cs

```diff
@@ -0,0 +1,235 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.Loader;
+using System.Text;
+using System.Threading;
+
+namespace System.Reflection
+{
+    internal unsafe ref partial struct TypeNameParser
+    {
+        private ReadOnlySpan<char> _input;
+        private int _index;
+        private int _errorIndex; // Position for error reporting
+
+        private Func<AssemblyName, Assembly?>? _assemblyResolver;
+        private Func<Assembly?, string, bool, Type?>? _typeResolver;
+        private bool _throwOnError;
+        private bool _ignoreCase;
+        private bool _extensibleParser;
+        private void* _stackMark;
```

Comment:
Why not make it a `ref` field? The value points to the stack so it wouldn't change anything; just a stylistic change.

Reply:
StyleCop is not happy with the `ref StackCrawlMark` field.

Also, I will need to rework this part to avoid perf regressions. The stack crawling needs to go through more stack frames in the current change that makes things quite a bit slower.